src/transformers/models/trocr/processing_trocr.py

```diff
@@ -70,21 +85,21 @@ def __call__(self, *args, **kwargs):
         """
         # For backward compatibility
         if self._in_target_context_manager:
-            return self.current_processor(*args, **kwargs)
-
-        images = kwargs.pop("images", None)
-        text = kwargs.pop("text", None)
-        if len(args) > 0:
-            images = args[0]
-            args = args[1:]
+            return self.tokenizer(*args, **kwargs)
```

Comment:
As far as I see it was an image processor, not a tokenizer, wasn't it?

Reply:
I think this is correct. This path is only taken if we are inside the `as_target_processor` context manager. So I simplified it - while keeping backward compatibility - by removing self.current_processor.

Reply:
Got it, I would leave it as it was, otherwise, `@as_target_processor` has no effect to `current_processor`. 

I'm not even sure it is used somewhere, probably we should deprecate `as_target_processor (I might be missing smth). @molbap do you know if it is used somewhere?

Reply:
I have put back `current_processor`. There is already a warning inside the context manager that it will be removed in transformers v5.

Reply:
Thanks for putting it back! Indeed, I missed the warning 😄